src/state/routing/slice.ts

```diff
@@ -115,7 +115,7 @@ export const routingApi = createApi({
             )
           }
 
-          return { data: result.data as GetQuoteResult }
+          return { data: (result.data || {}) as GetQuoteResult }
```

Comment:
should you return undefined instead?

probably not type safe to cast `{} as GetQuoteResult` given `GetQuoteResult` has required props?

Reply:
It's currently returning undefined, and that's causing a bunch of console errors on networks where this query fails. (BNB) Though, no observable ux issues afaict.

Reply:
The way I'm reading the error, I think returning `undefined` **and** an `error` value would work too, but I wasn't sure about that and this seemed like the smallest change.

Reply:
<img width="1123" alt="image" src="https://user-images.githubusercontent.com/5773490/222748448-98b7b03b-8fa0-4134-9ff1-ab8bea06b32e.png">

Reply:
what do you think of this? 

```suggestion
          if (result.data) {
            return { data: result.data as GetQuoteResult }
          }
          return { data: undefined, error: result.error as FetchBaseQueryError }
```